### PR TITLE
upgrade pylint and enable linting for PY3 code

### PR DESCRIPTION
**Pull request by clarketm**
Since *most* of our python is upgraded from PY2 -> PY3 (#13164), it is a good time to start linting it with a `python3` `pylint` linter.

> NOTE: due to limitations with `pip_import` in [`bazelbuild/rules_python`](https://github.com/bazelbuild/rules_python/issues/220) (e.g. bazelbuild/rules_python#220), ensure system `python` is linked to `python3` in your development environment.

---

To get this working there is a dependency on bazelbuild/rules_python#220 getting `pip_import` rule to be dynamic or a less appealing option is we can simply update our `pull-test-infra-bazel` image (e.g. `launcher.gcr.io/google/bazel:0.27.0`) to have a system `python` symlinked to `python3`. 

> May be worth trying out [rules_pip](https://github.com/apt-itude/rules_pip) as pointed out in bazelbuild/rules_python#220

### CODE REVIEW

**Comment by clarketm** [comment]
@fejta - Since this PR mutates the `WORKSPACE` file, shouldn't the `pull-test-infra-bazel` test perform a `bazel sync`. This is needed to properly test this PR.

File: triage/summarize.py [hunk 1]
```diff
@@ -484,6 +484,7 @@ def render(builds, clustered):
 
 SIG_LABEL_RE = re.compile(r'\[sig-([^]]*)\]')
 
+# pylint: disable=too-many-branches
```
Comment (fejta) [inline triage/summarize.py]: Doesn't this disable this lint for the whole file? If its just annotate_owners isn't it more appropriate to inline it there?

Comment (clarketm) [reply triage/summarize.py]: Oops. My intention was to **disable** only for `annotate_owners` function. Let me update to an inline comment.

**Pull request merged**